File: EVM-frontend/src/app/core/services/asignacion-actividad.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {
  AsignacionActividadRequest,
  AsignacionActividadResponse,
  UsuarioDisponibleResponse,
} from '../models/asignacion.models';

const API_URL = 'http://localhost:8081/api';

@Injectable({ providedIn: 'root' })
export class AsignacionActividadService {
  private http = inject(HttpClient);

  listarPorActividad(idActividad: number): Observable<AsignacionActividadResponse[]> {
    return this.http.get<AsignacionActividadResponse[]>(
      `${API_URL}/actividades/${idActividad}/asignaciones`
    );
  }

  listarUsuariosDisponibles(idActividad: number): Observable<UsuarioDisponibleResponse[]> {
    return this.http.get<UsuarioDisponibleResponse[]>(
      `${API_URL}/actividades/${idActividad}/usuarios-disponibles`
    );
  }

  asignar(
    idActividad: number,
    dto: AsignacionActividadRequest
  ): Observable<AsignacionActividadResponse[]> {
    return this.http.post<AsignacionActividadResponse[]>(
      `${API_URL}/actividades/${idActividad}/asignaciones`,
      dto
    );
  }

  retirar(idAsignacion: number): Observable<AsignacionActividadResponse> {
    return this.http.patch<AsignacionActividadResponse>(
      `${API_URL}/asignaciones/${idAsignacion}/retirar`,
      {}
    );
  }
}

File: EVM-backend/src/main/java/com/evm/backend/service/AsignacionActividadService.java
package com.evm.backend.service;

import com.evm.backend.dto.AsignacionActividadRequestDTO;
import com.evm.backend.dto.AsignacionActividadResponseDTO;
import com.evm.backend.dto.UsuarioDisponibleResponseDTO;

import java.util.List;

public interface AsignacionActividadService {

    List<AsignacionActividadResponseDTO> listarPorActividad(Long idActividad);

    List<UsuarioDisponibleResponseDTO> listarUsuariosDisponibles(Long idActividad);

    List<AsignacionActividadResponseDTO> asignar(Long idActividad, AsignacionActividadRequestDTO dto);

    AsignacionActividadResponseDTO retirar(Long idAsignacion);
}
